src/components/EpisodePreview/EpisodePreview.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { useContext } from "react";
+import {
+	EpisodesContext as Episodes
+} from "COMPONENTS/Episodes";
+
+export default function EpisodePreview(props){
+
+	// CONTEXT
+	//-----------------------------
+	const {
+		episodeCount, 
+		activeIndex,
+		setActiveIndex,
+		overlayActive,
+		setOverlayActive
+	} = useContext(Episodes);
+
+
+	// RENDER
+	// ----------------------------
+	const { 
+		index,
+		number: episodeNo,
+		title,
+		image: {
+			src: thumbSrc,
+			description: thumbDesc
+		},
+		video: {
+			src: videoSrc
+		}
+	} = props;
+
+	const previewId = `episode__${episodeNo}__preview`;
+
+	// handle visibility of active episode and episode preview overlay
+	const isActiveEpisode 	= activeIndex === index;
+	const thisOverlayActive = isActiveEpisode && overlayActive;
+	const hidden 			= !thisOverlayActive;
+
+	const isFirst 	= index === 0;
+	const isLast 	= index === episodeCount - 1;
+
+	const prevEpisodeNo = episodeNo - 1;
+	const nextEpisodeNo = episodeNo + 1;
+
+	function handleNavPressed(e, newIndex){
+		e.preventDefault();
+
+		setActiveIndex(newIndex);
+	}// handleNavPressed
+
+
+	function showOverlay(e, show){
+		e.preventDefault();
+
+		setOverlayActive(show);
+	}// showOverlay
+
+	return (
+		<aside>
+			<a
+				href={ videoSrc }
+				aria-controls={ previewId }
+				aria-label="Watch episode preview."
+				onClick={ (e) => { showOverlay(e, true) } }
+			>
+				<img
+					src={ thumbSrc }
+					alt={ thumbDesc }
+				/>
+			</a>
+			<div
+				id={ previewId }
+				aria-hidden={ hidden }
+			>
+				<header>
+					<h2>
+						{ `Episode ${episodeNo}` } 
+					</h2>
+					<h1>
+						{ title }
+					</h1>
+					<nav>
+						{ !isFirst && (
+							<a 
+								href={ `#episode__preview__${prevEpisodeNo}` }
+								aria-label="Previous."
+								onClick={ (e) => { 
+									handleNavPressed(e, index-1) 
+								}}
+							>
+								Prev
+							</a>
+						)}
+						{ !isLast && (
+							<a 
+								href={ `#episode__preview__${nextEpisodeNo}` } 
+								onClick={ (e) => { 
+									handleNavPressed(e, index+1) 
+								}}
+							>
+								Next
+							</a>
+						)}
+						<a 
+							href={ `#episode__details__${episodeNo}` }
+							onClick={ e => showOverlay(e, false) }
```

Comment:
So, more naming stuff - anything with a/b shouldn't take an argument - ie `showOverlay` implies the existence of `hideOverlay` and so you'd expect those functions to be called like `showOverlay()` and `hideOverlay()` - if it's one function that changes between two states then that'd be `toggleOverlay()`, but if it's based on a given argument then that's gotta be `setOverlay(true||false)`, right?

In this case, what I think we want is `setOverlay` and some binding to give us the `showOverlay` and `hideOverlay`.  BUT, we've already been given the terminlogy with the word `close` - therefore instead of `show`/`hide` we should be using `open`/`close` - ie:

```
function setOverlay(active, event){
    if(event) event.preventDefault();
    setOverlayActive(active);
}//setOverlay

const openOverlay = setOverlay.bind(this, true);
const closeOverlay = setOverlay.bind(this, false);

...

<a onClick={openOverlay}>
    Open
</a>

<a onClick={closeOverlay}>
   Close
</a>
```

The reason for `if(event) event.preventDefault()` is that it will use the `preventDefault()` when we use it for onClick events, but we are still able to use them as util-style functions with `setOverlay(false)` or even just `openOverlay()`.

Reply:
Cool yeah, makes sense.